### PR TITLE
Deprecate Mirror

### PR DESCRIPTION
**Pull request by JetRunner**
The mirror option for downloading is now deprecated due to the switching to Git LFS and removal of Tuna.

https://github.com/tuna/issues/issues/937

### CODE REVIEW

**Review by julien-c — APPROVED** [review]

**Comment by julien-c** [comment]
(i am not a maintainer, please wait for maintainer review/approval before merging!) @sgugger @LysandreJik

**Comment by JetRunner** [comment]
> (i am not a maintainer, please wait for maintainer review/approval before merging!) @sgugger @LysandreJik 

Oh I'll be more cautious next time (though I'm sure this one is okay)

File: src/transformers/file_utils.py
Comment (sgugger) [inline src/transformers/file_utils.py:1231]: Please use a proper deprecation warning here like in other parts of the codebase:
```
warnings.warn(..., FutureWarning)
```
as a log can be silenced and will be displayed every time this code is executed.

File: src/transformers/file_utils.py
Comment (sgugger) [inline src/transformers/file_utils.py:1234]: It looks like the `mirror` argument is now used in a different way, which is a breaking change (passing `mirror="tuna"` used to work and doesn't work anymore for instance). We don't do breaking changes before a major release, so this will need to be reverted.

**Review by sgugger — COMMENTED** [review]
Thanks for the PR and the concept of deprecating this `mirror` argument sounds good but:
- we use proper deprecation warnings for this
- we never do breaking changes between minor releases, so the argument should still behave in the same way for now with the deprecation warning, and we should only change its behavior or remove it for the next major version.

I think the easiest is to revert this merge and start fresh on a new PR.

**Comment by JetRunner** [comment]
> Thanks for the PR and the concept of deprecating this `mirror` argument sounds good but:
> 
> - we use proper deprecation warnings for this
> 
> - we never do breaking changes between minor releases, so the argument should still behave in the same way for now with the deprecation warning, and we should only change its behavior or remove it for the next major version.
> 
> 
> 
> I think the easiest is to revert this merge and start fresh on a new PR.

Hi @sgugger 

I was in a hurry (gonna be on a trip soon). Sorry for rushing for it!

It's not an actual breaking change because the HF mirrors have already been removed from these servers. They are already broken now. The warning is a way to explain why it breaks.

Could you point me to the correct way to throw a deprecated message?

**Comment by nreith** [comment]
Thanks @julien-c !
I just sent an email. Can you clarify how this would differ from the self-hosted option above, and when the self-hosted option would be available?

**Pull request merged**
